src/Commands/Server/ToolOperations.cs

```diff
@@ -75,6 +74,22 @@ private async ValueTask<CallToolResponse> OnCallTools(RequestContext<CallToolReq
             };
         }
 
+        if (_toolCommands == null)
+        {
+            var content = new Content
+            {
+                Text = "Tool commands not initialized. Call list tools first.",
+            };
+
+            _logger.LogWarning(content.Text);
+
+            return new CallToolResponse
+            {
+                Content = [content],
+                IsError = true,
+            };
+        }
+
```

Comment:
I addressed this in another PR.  You can remove this.

Reply:
Undid changes for this file.